### PR TITLE
feat: multiple new functionalities

### PR DESCRIPTION
**Pull request by jonymaster**
Sorry mate, I screwed up and did not create a branch for each change. Will do next time.

Each commit is for something I added, please review when you have time. Thanks

### CODE REVIEW

File: mobileiron_api/api/base_api.py [hunk 1]
```diff
@@ -50,9 +52,9 @@ def _execute_call(self,
                       params: str = None,
                       json_value: object = None):
         response = None
+        if params is not None:
+            url = f"{url}?{params}"
```
Comment (lawiet47) [inline mobileiron_api/api/base_api.py:56]: This we only need in GET methods, no?

File: mobileiron_api/api/base_api.py [hunk 2]
```diff
@@ -50,9 +52,9 @@ def _execute_call(self,
...
         if http_method == 'get':
```
Comment (lawiet47) [inline mobileiron_api/api/base_api.py]: Does this block serve a purpose

Comment (jonymaster) [reply mobileiron_api/api/base_api.py]: it was included only in 'get' method, so I put it before going through the method selection

**Review by lawiet47 — CHANGES_REQUESTED** [review]
.

File: mobileiron_api/api/base_api.py (hunk 1 above)
Comment (jonymaster) [reply mobileiron_api/api/base_api.py:56]: We need it also in put and post

File: mobileiron_api/api/base_api.py [hunk 3]
```diff
@@ -51,20 +51,11 @@ def _execute_call(self,
...
-        elif http_method == 'post':
-            response = requests.post(url, headers=headers, json=json_value, timeout=self._requests_timeout)
-        elif http_method == 'put':
-            if params is not None:
-                response = requests.put(url, headers=headers, params=params, json=json_value, timeout=self._requests_timeout)
-            else:
-                response = requests.put(url, headers=headers, json=json_value, timeout=self._requests_timeout)
-        elif http_method == 'delete':
-            response = requests.delete(url, headers=headers, json=json_value, timeout=self._requests_timeout)
+        response = method(url, headers=headers, params=params if http_method != 'get' else None, json=json_value,
```
Comment (lawiet47) [inline mobileiron_api/api/base_api.py:57]: Did you test it locally?

Comment (jonymaster) [reply mobileiron_api/api/base_api.py:57]: yes, I tested get, delete, post and put

**Pull request merged**